utils/authOptions.ts

```diff
@@ -0,0 +1,120 @@
+      name: 'Credentials',
+      credentials: {
+        email: { label: 'Email', type: 'email' },
+        password: { label: 'Password', type: 'password' },
+        cfTurnstileResponse: { label: 'Token', type: 'text' },
+      },
+      async authorize(credentials) {
+        // Replace this with your actual login API call or database check.
+        const res = await fetch(`${baseUrl}/content/login`, {
+          method: 'POST',
+          headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            email: credentials?.email,
+            password: credentials?.password,
+            cfTurnstileResponse: credentials?.cfTurnstileResponse,
+          }),
+        });
+
+        if (!res.ok) {
+          // eslint-disable-next-line no-console
+          console.error(`Failed to authenticate: ${res.status} ${res.statusText}`);
+          return null;
+        }
+
+        const data = await res.json();
+
+        // If login is successful, return the user object.
+        if (res.ok) {
+          const { user } = data.data;
+          // If we get here, we have a successful login
+          // Return the user object that will be saved in the token
+          return {
+            id: user.id,
+            name: user.name,
+            email: user.email,
+            role: user.role,
+            image: user.image,
+            accessToken: data.data.access_token, // If your API returns a token
+          };
+        }
+        // Return null if login failed.
+        return null;
+      },
+    }),
+  ],
+  // callbacks: {
+  //   async jwt({ token, user }) {
+  //     if (user) {
+  //       token.accessToken = user.accessToken;
+  //       token.id = user.id;
+  //       token.name = user.name;
+  //       token.email = user.email;
+  //     }
+  //     return token;
+  //   },
+
+  //   async session({ session, token }) {
+  //     // Add the access token and user ID to the session
+
+  //     session.user.id = token.id as string;
+  //     session.accessToken = token.accessToken as string;
+  //     session.user.email = token.email as string;
+  //     session.user.name = token.name as string;
+
+  //     return session;
+  //   },
+  // },
+  callbacks: {
+    async jwt({ token, user }) {
+      const newToken = { ...token };
+      if (user) {
+        newToken.accessToken = user.accessToken;
+        newToken.id = user.id;
+        newToken.name = user.name;
+        newToken.email = user.email;
+      }
+      return newToken;
+    },
+
+    async session({ session, token }) {
+      const newSession = { ...session };
+      newSession.accessToken = token.accessToken;
+      newSession.user.id = token.id;
+      newSession.user.email = token.email;
+      newSession.user.name = token.name;
+      return newSession;
+    },
+  },
+  pages: {
+    signIn: '/control-panel/login', // Custom login page path
```

Comment:
how can we expand this to work with the admin-panel and platform as well?

Reply:
I have removed this for now, since it is not mandatory for configuration.
This is mainly used to customize the default authentication pages provided by NextAuth.

As far as i have researched, it is possible to make this reusable too, but i suggest to get back to this once we have finished with the middleware and the whole auth process.